src/pages/workspaces/workspace/Data.js

```diff
@@ -360,14 +364,37 @@ const ReferenceDataContent = ({ workspace: { workspace: { namespace, attributes
   ])
 }
 
-const ToolDrawer = withModalDrawer()(({
-  workspace, workspace: { workspace: { workspaceId } }, onDismiss, onIgvSuccess, entityMetadata, entityKey, selectedEntities
+const ToolDrawer = withModalDrawer()(requesterPaysWrapper({
+  onDismiss: ({ onDismiss }) => onDismiss()
+})(({
+  workspace, workspace: { workspace: { workspaceId } }, onDismiss, onIgvSuccess,
+  onRequesterPaysError, entityMetadata, entityKey, selectedEntities
 }) => {
+  const { workspace: { bucketName, name: wsName, namespace } } = workspace
+
   const [toolMode, setToolMode] = useState()
+  const [notebookNames, setNotebookNames] = useState()
+  const signal = useCancellation()
+
+  const { Buckets } = Ajax(signal)
+
+  const loadNotebookNames = _.flow(
+    withRequesterPaysHandler(onRequesterPaysError),
+    withErrorReporting('Error loading notebooks')
+  )(async () => {
+    const notebooks = await Buckets.listNotebooks(namespace, bucketName)
+    // slice removes 'notebooks/' and the .ipynb suffix
+    setNotebookNames(notebooks.map(notebook => notebook.name.slice(10, -6)))
+  })
+  Utils.useOnMount(() => {
+    loadNotebookNames()
```

Comment:
If this is only used here, our convention is to move the definition inside of the `useOnMount` as well, for clarity.

Reply:
Done